src/rdf4cpp/rdf/datatypes/xsd/Name.hpp

```diff
@@ -0,0 +1,50 @@
+
+/**
+ * @file Registers xsd:Name with DatatypeRegistry
+ */
+
+#ifndef RDF4CPP_XSD_NAME_HPP
+#define RDF4CPP_XSD_NAME_HPP
+
+#include <rdf4cpp/rdf/datatypes/registry/DatatypeMapping.hpp>
+#include <rdf4cpp/rdf/datatypes/registry/LiteralDatatypeImpl.hpp>
+
+#include <cstdint>
+#include <ostream>
+#include <regex>
+
+namespace rdf4cpp::rdf::datatypes::registry {
+/*
+ * Name of the datatype. This is kept so that we won't need to type it over and over again.
+ */
+constexpr static registry::ConstexprString xsd_name{"http://www.w3.org/2001/XMLSchema#name"};
+
+/**
+ * Defines the mapping between the LiteralDatatype IRI and the C++ datatype.
+ */
+template<>
+struct DatatypeMapping<xsd_name> {
+    using cpp_datatype = std::string;
+};
+
+/**
+ * Specialisation of from_string template function.
+ */
+template<>
+inline LiteralDatatypeImpl<xsd_name>::cpp_type LiteralDatatypeImpl<xsd_name>::from_string(std::string_view s) {
+    const std::regex non_name_regex("(\\n)|(\\t)|(\\r)|([ ]{2,})|(^[ ].*)|(.*[ ]$)");
```

Comment:
double check if the regex is fine and covers all cases. Add comments that explain what happens here